slang.h

```diff
@@ -2098,6 +2098,21 @@ extern "C"
         SlangReflectionType* const* specializationArgs,
         ISlangBlob**                outDiagnostics);
 
+        /// Get the number of hashed strings
+    SLANG_API SlangUInt spReflection_getHashedStringCount(
+        SlangReflection*  reflection);
+
+        /// Get a hashed string. The number of chars is writtent in outCount. Note the count does *NOT* including terminating 0,
+        /// and the returned pointer will not generally have a terminating zero.
+    SLANG_API const char* spReflection_getHashedString(
+        SlangReflection*  reflection,
+        SlangUInt index,
+        size_t* outCount);
+
+        /// Calculate a string hash.
+        // Count should *NOT* include terminating zero.
+    SLANG_API int spCalcStringHash(const char* chars, size_t count);
```

Comment:
Aside: I'm still on the fence about how I feel about "calc" as a term in our internal naming conventions, and seeing it creep into public API definitely makes me even more uneasy. I'd be more comfortable with `spGetStringHash` or `spComputeStringHash`, but I'm not going to insist that we change anything.

Reply:
One more note here: what is the trade-off between having `spCalcStringHash` as a stand-alone utility (callable with any string) vs. having `spReflection_getHashedString` return both the string data and its hash code?